src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2069,21 +2070,21 @@
   <Node Name="BoundImplicitIndexerAccess" Base="BoundExpression" SkipInNullabilityRewriter="true" HasValidate="true">
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow" />
 
+    <!-- The target of the index operation -->
+    <Field Name="Receiver" Type="BoundExpression" Null="disallow" />
```

Comment:
I'm very glad we're able to move back to this design with the confidence that the ValueCheck logic only needs to be abstracted for placeholders that might be "this" (not as bad as we feared). 
Thanks a lot!